src/components/forms/formComponents.js

```diff
@@ -0,0 +1,387 @@
+    const formValueClone = cloneDeep(formValue);
+    formValueClone.splice(index, 1);
+    setFormValue(formValueClone);
+    onChange({key: id, value: formValueClone, isValid: true});
+  };
+
+  return div({}, [
+    div({
+      className: 'formControl-group flex-row',
+    }, [
+      input({
+        id,
+        type: 'text',
+        className: `form-control ${error ? 'errored' : ''}`,
+        placeholder: placeholder || title,
+        style: { ...styles.inputStyle, ...inputStyle },
+        disabled,
+        'aria-describedby': ariaDescribedby,
+        onKeyUp: (event) => event.code === 'Enter' ? pushValue(event.target) : setError(),
+        onFocus: () => setError()
+      }),
+      h(button, {
+        className: 'form-btn btn-xs',
+        type: 'button',
+        disabled,
+        style: {
+          marginTop: 0,
+          minWidth: 'fit-content'
+        },
+        onClick: () => pushValue(document.getElementById(id))
+      }, [
+        span({
+          className: 'glyphicon glyphicon-plus',
+          'aria-label': 'Add',
+          style: { margin: '0 8px' },
+          isRendered: !disabled
+        })
+      ])
+    ]),
+    errorMessage(error),
+    div({ className: 'flex-row', style: { justifyContent: null } },
+      formValue.map((val, i) => {
+        return h(button, {
+          key: val,
+          className: 'pill btn-xs',
+          type: 'button',
+          disabled,
+          onClick: () => removePill(i)
+        }, [
+          val,
+          span({
+            className: 'glyphicon glyphicon-remove',
+            style: { marginLeft: '8px' },
+            isRendered: !disabled
+          })
+        ]);
+      })
+    )
+  ]);
+};
+
+// Using react-select/creatable - Passing config directly through!
+export const formInputSelect = (config) => {
+  const {
+    id, title, disabled, required, error, setError,
+    selectOptions, placeholder, ariaDescribedby,
+    formValue, isCreatable, isMulti, setFormValue,
+    selectConfig = {}
+  } = config;
+
+  const component = (isCreatable ? Creatable : Select);
+
+  const isStringArr = isString(selectOptions[0]);
+  const normalizedOptions = isStringArr
+    ? selectOptions.map((option) => { return {key: option, displayValue: option }; })
+    : selectOptions;
+
+  return h(component, {
+    key: id,
+    id,
+    isClearable: true, //ensures that selections can be cleared from dropdown, adds an 'x' within input box
+    isMulti,
+    required,
+    isDisabled: disabled,
+    placeholder: placeholder || `Search for ${title}...`,
+    className: `form-select ${error ? 'errored' : ''}`,
+    onChange: (option) => {
+      if (isStringArr) {
+        if (isMulti) {
+          // string result, multiple options
+          onFormInputChange(config, option.map((o) => o.displayValue));
+          setFormValue(option);
+          return;
+        }
+        // string result, only one option
+        onFormInputChange(config, option.displayValue);
```

Comment:
When I click the 'x' button, onChange is triggered and I receive this error:
![Screenshot (105)](https://user-images.githubusercontent.com/81024249/187959853-634b469e-8abe-4341-9401-d7934d5bdd1d.png)